src/template.rs

```diff
@@ -279,6 +280,7 @@ impl Template {
 
         let mut buffer: String = String::new();
         let mut state = ParserState::Text;
+        let mut old_state = ParserState::Text;
```

Comment:
Do we need to maintain old_state here? Perhaps we can just reset to `ParserState::Text` on exit, like helper.